pkg/controller/petset/pet.go

```diff
@@ -251,8 +251,8 @@ func (p *apiServerPetClient) SyncPVCs(pet *pcb) error {
 		if !exists {
 			if err := p.createPVC(&pet.pvcs[i]); err != nil {
 				errMsg += fmt.Sprintf("Failed to create %v: %v", pvc.Name, err)
+				p.event(pet.parent, "Create", fmt.Sprintf("pvc: %v", pvc.Name), err)
 			}
-			p.event(pet.parent, "Create", fmt.Sprintf("pvc: %v", pvc.Name), err)
```

Comment:
apiServerPetClient implements an interface, you don't want to impose that all callers of that interface understand how to classify the error as IsNotFound. The alternative is that we return the same type of http 404 error from the the interface, even it it's backend by local cache instead of apiserver, which is gross.